### PR TITLE
[Bugfix] Exclude violation `operation.notAllowed` with 405

### PR DESCRIPTION
**Pull request by pboos**
When there is a `validation.request.operation.notAllowed` violation and the response already is `405` (Method Not Allowed) then this should not be reported as a violation.

Reason: The API responds correctly that this method is not allowed. There is nothing to be fixed in the API nor the spec.

### CODE REVIEW

File: openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/exclusions/InternalViolationExclusions.java [hunk 1]
```diff
@@ -36,4 +37,9 @@ private boolean falsePositive404(OpenApiViolation violation) {
     private boolean falsePositive400(OpenApiViolation violation) {
         return violation.getDirection() == Direction.REQUEST && violation.getResponseStatus().orElse(0) == 400;
     }
+
+    private boolean falsePositive405(OpenApiViolation violation) {
+        return violation.getResponseStatus().orElse(0) == 405
+            && "validation.request.operation.notAllowed".equals(violation.getRule());
```
Comment (huguenin) [inline openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/exclusions/InternalViolationExclusions.java:43]: Should the rule name here go into a constant? Are all rules defined somewhere?

**Review by huguenin — APPROVED** [review]
Change looks good, just a formality. Also, there are tons of changes that seem unrelated to the fix - I didn't check those in detail. Would recommend not including these in the future, or creating a separate PR.

File: openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/exclusions/InternalViolationExclusions.java (hunk 1 above)
Comment (pboos) [reply openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/exclusions/InternalViolationExclusions.java]: They are not defined anywhere in `swagger-request-validator`. I have now extracted them here in the file at least.

Comment (pboos) [reply openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/exclusions/InternalViolationExclusions.java]: Will create a const file now as we also use them in other places.

**Pull request merged**